### ISSUE TITLE
How to pass target to Blazor.platform.callMethod.

### ISSUE THREAD

**Issue by mousehuang** [aspnet/Blazor#855]
For the Blazor.platform.callMethod, the second parameter is a Target. It can be null to invoking a static method. But I don't know how to pass an object instance to it and invoking the instance method.

Blazor.platform.callMethod(
                    logToConsoleMethod,         // Method handle
                    null,                       // Target // HOW TO PASS AN OBJECT INSTANCE to IT.
                    [ messageAsDotNetString ]   // Arguments
                );

**Comment by Suchiman** [comment]
By just passing an object 😄 that easy.
Ok, here's the explanation: Any managed method that returns an object reference (which is just a pointer) will be marshalled as number (the pointer address) into JS which you can then pass back into managed code where an object reference is expected (so the number which is the object pointer address is converted back into an object reference).
Example:
```js
var obj = Blazor.platform.callMethod(objReturningMethod, null, [...]);
Blazor.platform.callMethod(logToConsoleMethod, obj, ...);
```

**Comment by SteveSandersonMS** [comment]
@Suchiman is correct.

Longer term, we're aiming to make it simpler to pass .NET references to JS and vice-versa, and make it safe to hold those references across async operations.

Currently if you pass a .NET object reference to JS then you **must only** use it during the same synchronous call. You **must not** store the reference and attempt to use it later, after having yielded the thread (and if you do so, you will randomly get crashes because pointers are not necessarily still valid after the .NET side performs a garbage collection pass).

So in summary, this is very advanced and until we implement friendlier and safer APIs for it, you probably shouldn't do it. In the short term a safer alternative is only to exchange IDs between JS and .NET (e.g., ints or strings).
